sdk/typescript/src/signature-mpc/encrypt_user_share.ts

```diff
@@ -37,11 +54,105 @@ export const sendUserShareToSuiPubKey = async (
 		new Uint8Array(dwallet.secretKeyShare),
 		recipientData?.encryptionKey!,
 	);
+
 	return await transferEncryptedUserShare(
 		client,
 		keypair,
 		encryptedUserShareAndProof,
 		activeEncryptionKeyObjID,
-		dwallet.dwalletId,
+		dwallet,
+		signedDWalletPubKeys,
+	);
+};
+
+export type EncryptedUserShare = {
+	dwalletId: string;
+	encryptedUserShareAndProof: number[];
+	encryptionKeyObjID: string;
+	signedDWalletPubkeys: number[];
+	senderPubKey: number[];
+};
+
+export const getEncryptedUserShareByObjID = async (
+	client: DWalletClient,
+	objID: string,
+): Promise<EncryptedUserShare | null> => {
+	const response = await client.getObject({
+		id: objID,
+		options: { showContent: true },
+	});
+
+	const objectFields =
+		response.data?.content?.dataType === 'moveObject'
+			? (response.data?.content?.fields as unknown as {
+					dwallet_id: string;
+					encrypted_secret_share_and_proof: number[];
+					encryption_key_id: string;
+					signed_dwallet_pubkeys: number[];
+					sender_pubkey: number[];
+			  })
+			: null;
+
+	return objectFields
+		? {
+				dwalletId: objectFields.dwallet_id,
+				encryptedUserShareAndProof: objectFields.encrypted_secret_share_and_proof,
+				encryptionKeyObjID: objectFields.encryption_key_id,
+				signedDWalletPubkeys: objectFields.signed_dwallet_pubkeys,
+				senderPubKey: objectFields.sender_pubkey,
+		  }
+		: null;
+};
+
+export const acceptUserShare = async (
+	encryptedUserShare: EncryptedUserShare,
+	expectedSourceSuiAddress: string,
+	encryptionKey: Uint8Array,
+	decryptionKey: Uint8Array,
+	dwalletID: string,
+	encryptionKeysHolderObjID: string,
+	client: DWalletClient,
+	keypair: Keypair,
+): Promise<boolean> => {
+	let dwallet = await getDwalletByObjID(client, dwalletID);
+	let publicKey = new Ed25519PublicKey(encryptedUserShare?.senderPubKey!);
+	let serializedPubkeys = serialized_pubkeys_from_decentralized_dkg_output(
+		new Uint8Array(dwallet?.decentralizedDKGOutput!),
+	);
+	if (
+		!(await publicKey.verify(
+			serializedPubkeys,
+			new Uint8Array(encryptedUserShare?.signedDWalletPubkeys!),
+		))
+	) {
+		return false;
+	}
+	if (publicKey.toSuiAddress() !== expectedSourceSuiAddress) {
+		return false;
+	}
+
+	const decryptedKeyShare = decrypt_user_share(
+		encryptionKey,
+		decryptionKey,
+		new Uint8Array(encryptedUserShare?.encryptedUserShareAndProof!),
+	);
+
+	if (!verify_user_share(decryptedKeyShare, new Uint8Array(dwallet?.decentralizedDKGOutput!))) {
+		return false;
+	}
+	let dwalletToSend = {
+		dwalletId: dwalletID,
+		secretKeyShare: Array.from(decryptedKeyShare),
+		decentralizedDKGOutput: dwallet!.decentralizedDKGOutput,
+	};
+
+	await sendUserShareToSuiPubKey(
```

Comment:
```suggestion
	// Encrypt it to self, so that in the future we'd know that we already verified everything and only need to verify our signature.
	await sendUserShareToSuiPubKey(
```
This code is very lacking in documentation, please add